src/Controller/PictureController.php

```diff
@@ -0,0 +1,109 @@
+
+use App\Entity\Picture;
+use App\Entity\Villa;
+use App\Form\PictureType;
+use App\Repository\PictureRepository;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+
+/**
+ * @Route("/picture")
+ */
+class PictureController extends AbstractController
+{
+    /**
+     * @Route("/", name="picture_index", methods={"GET"})
+     */
+    public function index(PictureRepository $pictureRepository): Response
+    {
+        return $this->render('picture/index.html.twig', [
+            'pictures' => $pictureRepository->findAll(),
+        ]);
+    }
+
+    /**
+     * @Route("/new", name="picture_new", methods={"GET","POST"})
+     */
+    public function new(Request $request): Response
+    {
+        $picture = new Picture();
+        $form = $this->createForm(PictureType::class, $picture);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $entityManager = $this->getDoctrine()->getManager();
+            $file=$form->get('photoFile')->getData();
+            $originalFilename = pathinfo($file->getClientOriginalName(), PATHINFO_FILENAME);
+            $number=1;
+            $newFilename=$originalFilename.$number;
+            while (file_exists($this->getParameter('pictures_directory'). '/' .$newFilename)) {
+                $number++;
+                $newFilename=$originalFilename.$number;
+            }
+            $picture->setPhoto("/pictures/".$newFilename);
+            $file->move(
+                $this->getParameter('pictures_directory'),
+                $newFilename
+            );
+            $entityManager->persist($picture);
+            $entityManager->flush();
+
+            return $this->redirectToRoute('picture_index');
+        }
+
+        return $this->render('picture/new.html.twig', [
+            'picture' => $picture,
+            'form' => $form->createView(),
+        ]);
+    }
+
+    /**
+     * @Route("/{id}", name="picture_show", methods={"GET"}, requirements={"id"="\d+"})
+     */
+    public function show(Picture $picture): Response
+    {
+        return $this->render('picture/show.html.twig', [
+            'picture' => $picture,
+        ]);
+    }
+
+    /**
+     * @Route("/{id}/edit", name="picture_edit", methods={"GET","POST"})
+     */
+    public function edit(Request $request, Picture $picture): Response
+    {
+        $form = $this->createForm(PictureType::class, $picture);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $this->getDoctrine()->getManager()->flush();
+
+            return $this->redirectToRoute('picture_index');
+        }
+
+        return $this->render('picture/edit.html.twig', [
+            'picture' => $picture,
+            'form' => $form->createView(),
+        ]);
+    }
+
```

Comment:
Don't do make :crud for everything !!
this controller is probably not working at all.